### PR TITLE
Convert nix-bitcoin extraConfig options to regular options

### PR DESCRIPTION
**Pull request by jonasnick**

### CODE REVIEW

File: modules/bitcoind.nix [hunk 1]
```diff
@@ -209,6 +217,31 @@ in {
...
         description = "ZMQ address for zmqpubrawtx notifications";
       };
+      assumevalid = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "00000000000000000000e5abc3a74fe27dc0ead9c70ea1deb456f11c15fd7bc6";
+        description = "If this block is in the chain assume that it and its ancestors are valid and potentially skip their script verification";
+      };
+      addnode = mkOption {
+        type = types.nullOr types.str;
```
Comment (erikarvstedt) [inline modules/bitcoind.nix]: Would be nice if this were `addnodes` with type `listOf str`.
The list should be translated into multiple `addnode=` statements.

Comment (jonasnick) [reply modules/bitcoind.nix]: Oh, forgot to fix this. Sure. Added a commit.

File: modules/bitcoind.nix [hunk 2]
```diff
@@ -209,6 +217,34 @@ in {
...
+        example = "00000000000000000000e5abc3a74fe27dc0ead9c70ea1deb456f11c15fd7bc6";
+        description = "If this block is in the chain assume that it and its ancestors are valid and potentially skip their script verification";
+      };
+      addnodes = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        example = [ "ecoc5q34tmbq54wl.onion" ];
+        description = ''
+          Add nodes to connect to and attempt to keep the connections open. Will
+          add one addnode= entry to the config per array element.
```
Comment (erikarvstedt) [inline modules/bitcoind.nix]: Nit: `array` is `list` in nix lingo.
I think the implementation specifics here are obvious and don't warrant further explanation. So I'd remove the whole `Will add...` sentence.

File: modules/bitcoind.nix [hunk 3]
```diff
@@ -209,6 +217,34 @@ in {
...
+        '';
+      };
+      discover = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
```
Comment (erikarvstedt) [inline modules/bitcoind.nix]: Nit: single-line string like in `addresstype` is enough.

File: modules/bitcoind.nix [hunk 4]
```diff
@@ -227,10 +227,7 @@ in {
         type = types.listOf types.str;
         default = [];
         example = [ "ecoc5q34tmbq54wl.onion" ];
-        description = ''
-          Add nodes to connect to and attempt to keep the connections open. Will
-          add one addnode= entry to the config per array element.
-        '';
+        description = "Add nodes to connect to and attempt to keep the connections open. Will add one addnode= entry to the config per list element.";
```
Comment (erikarvstedt) [inline modules/bitcoind.nix]: For long descriptions multi-line is better :smiley:

Comment (jonasnick) [reply modules/bitcoind.nix]: Oops I mistook the line you were commenting on above. Fixed.

File: modules/bitcoind.nix [hunk 5]
```diff
@@ -209,6 +217,32 @@ in {
         example = "tcp://127.0.0.1:28333";
         description = "ZMQ address for zmqpubrawtx notifications";
       };
+      assumevalid = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "00000000000000000000e5abc3a74fe27dc0ead9c70ea1deb456f11c15fd7bc6";
+        description = "If this block is in the chain assume that it and its ancestors are valid and potentially skip their script verification";
```
Comment (erikarvstedt) [inline modules/bitcoind.nix]: Sorry for the nit. Wrap this long line and we're done. :cat:

**Comment by erikarvstedt** [comment]
Looks perfect now.
Maybe squash the last 2 commits.

**Pull request merged**